index.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+const cookieSession = require("cookie-session");
+const express = require("express");
+const expressHandlebars = require("express-handlebars");
+const githubAuth = require("connect-oauth-github");
+const github = require("github-request");
+const postRequest = require("request").post;
+const Slack = require("./lib/slack");
+const version = require("./package").version;
+const config = require("./config");
+
+const slackClient = new Slack(config.slack);
+
+const app = express();
+app.use(cookieSession({
+	name: "session",
+	keys: config.web.cookieKeys
+}));
+
+app.engine("handlebars", expressHandlebars({ defaultLayout: "main" }));
+app.set("view engine", "handlebars");
+
+const gha = githubAuth.createClient({
+	id: config.github.clientId,
+	secret: config.github.secret,
+	scope: "user:email"
+});
+
+app.get("/", (request, response) => {
+	response.render("index");
+});
+
+app.get("/auth", gha.handshake);
+
+app.get("/signup", gha.authorize, (request, response) => {
+	let ghaUser = gha.users[request.sessionID];
+
+	github.request({
+		path: "/user?access_token=" + ghaUser.accessToken,
+	}, (error, data) => {
+		if (error) {
+			// TODO: Implement error page
+			return response.status(500).render("error");
+		}
+
+		// TODO: Filter on data.login blacklist
```

Comment:
How do we want to manage this list? This is for cross-medium banning (i.e., if you get banned on GitHub, we can also ban you from joining Slack).